src/passes/GlobalStructInference.cpp

```diff
@@ -421,8 +422,9 @@ struct GlobalStructInference : public Pass {
           Expression* ret;
           if (value.isConstant()) {
             // This is known to be a constant, so simply emit an expression for
-            // that constant.
-            ret = value.getConstant().makeExpression(wasm);
+            // that constant, and handle if the field is packed.
+            auto* expr = value.getConstant().makeExpression(wasm);
```

Comment:
It might be worth considering a refactoring that would put the knowledge that we are representing an `i8` or `i16` value directly into the `Literal`, which would prevent users from having to remember and specially handle the case of packed values.

Reply:
Interesting idea. I think perhaps not on `PossibleConstantValues` which just means a wasm value, not necessarily one stored to a heap field. Adding this info there would add size + overhead in places we don't need it (some passes store very many such values, e.g. GUFA). But perhaps we could have `HeapStoredValue` or such. I'll keep thinking about it.